### PR TITLE
Accumulated changes from 2021

### PR DESCRIPTION
**Pull request by jrha**

### CODE REVIEW

File: makeweathermap.py [hunk 1]
```diff
@@ -321,6 +321,45 @@ def process_links(con, config, weathermap):
...
+    cur.execute("""
+        select hostname, ifName, ifDescr, ifAlias, ifSpeed, ports.device_id, port_id
+        from ports
+        join devices on ports.device_id = devices.device_id
+        where ports.ifAdminStatus = 'up' and (ports.ifName like 'vlan%' or ports.ifName like 'vni%')
+    """)
+    rows = cur.fetchall()
+
+    for row in rows:
+        raw_hostname, vlan, ifDescr, ifAlias, ifSpeed, device_id, port_id = row
```
Comment (Gargravarr2112) [inline makeweathermap.py:339]: Can be merged into the previous line

File: makeweathermap.py [hunk 2]
```diff
@@ -321,6 +321,45 @@ def process_links(con, config, weathermap):
...
+        #hostname = raw_hostname.split('.', 1)[0].replace(' ', '_').lower()
+        hostname = raw_hostname.replace(' ', '_').lower()
+        vlan = vlan.replace(' ', '_').lower()
+        #if not ifSpeed:
+        #    ifSpeed = 1 * G
```
Comment (Gargravarr2112) [inline makeweathermap.py:344]: How comes this is commented out?

**Review by Gargravarr2112 — APPROVED** [review]

**Review by Gargravarr2112 — APPROVED** [review]

**Review by Gargravarr2112 — APPROVED** [review]

**Review by Gargravarr2112 — COMMENTED** [review]
Merge into previous commit with same message?

**Review by Gargravarr2112 — COMMENTED** [review]
Is there a neater way to do this in Python than a chain of if/elif's?

**Review by Gargravarr2112 — APPROVED** [review]

File: makeweathermap.py [hunk 3]
```diff
@@ -23,6 +24,20 @@
...
+    return int(text) if text.isdigit() else text
+
+
+def natural_keys(text):
+    '''
+    alist.sort(key=natural_keys) sorts in human order
+    http://nedbatchelder.com/blog/200712/human_sorting.html
+    (See Toothy's implementation in the comments)
+    '''
+    return [ atoi(c) for c in re_split(r'(\d+)', text[0]) ]
```
Comment (Gargravarr2112) [inline makeweathermap.py:40]: text[0] implies only the first character of the string; from the context, is this working on objects? If so, 'text' is the wrong parameter name.

**Review by Gargravarr2112 — APPROVED** [review]

**Review by Gargravarr2112 — APPROVED** [review]

File: makeweathermap.py [hunk 4]
```diff
@@ -103,9 +103,14 @@ def process_links(con, config, weathermap):
...
-        where links.remote_hostname not like '%%.gridpp.rl.ac.uk' and links.remote_hostname not like '%%.fds.rl.ac.uk' and ifSpeed > %s and ifName not like 'ManagementEthernet%%'
-    """, M)
+        from links
+        join ports on ports.port_id=links.local_port_id
+        join devices on devices.device_id=ports.device_id
+        where links.remote_hostname not like '%%.gridpp.rl.ac.uk'
+            and links.remote_hostname not like '%%.fds.rl.ac.uk'
+            and ifSpeed > %s
+            and ifName not like 'ManagementEthernet%%'
+    """ % config.get('links', 'min_ifspeed'))
```
Comment (Gargravarr2112) [inline makeweathermap.py:175]: % is SQL-injectable, variables should be passed as parameters

Comment (Gargravarr2112) [reply makeweathermap.py:175]: Also % formatting is deprecated in Python3

File: makeweathermap.py [hunk 5]
```diff
@@ -110,6 +110,9 @@ def process_links(con, config, weathermap):
             and links.remote_hostname not like '%%.fds.rl.ac.uk'
             and ifSpeed > %s
             and ifName not like 'ManagementEthernet%%'
+            and remote_port not like 'ManagementEthernet%%'
+            and ifAlias not like '%%mgt%%'
+            and ifAlias not like '%%man%%'
```
Comment (Gargravarr2112) [inline makeweathermap.py:174]: Is this going to exclude links to VMs that bridge the two networks, e.g. manvpn2, mandns1, manradius...

Comment (jrha) [reply makeweathermap.py:174]: Maybe, but we don't want to map links to hosts anyway so it's a moot point.

**Review by Gargravarr2112 — APPROVED** [review]

**Review by Gargravarr2112 — APPROVED** [review]

**Review by Gargravarr2112 — APPROVED** [review]

**Pull request merged**